src/main/webapp/resources/js/pages/projects/samples-metadata-import/components/SampleMetadataImportMapHeaders.tsx

```diff
@@ -27,46 +31,99 @@ const { Text } = Typography;
+    getMetadataRestrictions().then((data) => {
+      setRestrictions(data);
+    });
+  }, []);
 
   const onSubmit = async () => {
-    if (projectId && column) {
+    if (projectId && updatedSampleNameColumn) {
       setLoading(true);
-      await dispatch(setSampleNameColumn({ projectId, column }));
+      await dispatch(
+        setSampleNameColumn({ projectId, updatedSampleNameColumn })
+      );
+      await dispatch(updateHeaders(updatedHeaders));
       navigate(`/${projectId}/sample-metadata/upload/review`);
     }
   };
 
+  const onSampleNameColumnChange = (value: string) => {
+    setUpdatedSampleNameColumn(value);
+  };
+
+  const onRestrictionChange = (item: MetadataHeaderItem, value: string) => {
+    const index = updatedHeaders.findIndex(
+      (header) => header.rowKey === item.rowKey
+    );
+    if (index !== -1) {
+      const updatedHeadersItem = { ...updatedHeaders[index] };
+      updatedHeadersItem.restriction = value;
+      updatedHeaders[index] = updatedHeadersItem;
+    }
+  };
+
+  const columns = [
+    {
+      title: i18n("SampleMetadataImportMapHeaders.table.header"),
+      dataIndex: "name",
+    },
+    {
+      title: i18n("SampleMetadataImportMapHeaders.table.restriction"),
+      dataIndex: "restriction",
+      render(id: number, item: MetadataHeaderItem) {
+        return (
+          <Radio.Group
+            options={restrictions}
+            defaultValue={item.restriction}
+            disabled={item.name === updatedSampleNameColumn}
+            onChange={({ target: { value } }) =>
+              onRestrictionChange({ ...item }, value)
+            }
+            optionType="button"
+          />
+        );
+      },
+    },
+  ];
+
   return (
     <SampleMetadataImportWizard current={1}>
       <Text>{i18n("SampleMetadataImportMapHeaders.description")}</Text>
-      <Radio.Group
-        style={{ width: `100%` }}
-        value={column}
-        onChange={(e: RadioChangeEvent) => setColumn(e.target.value)}
+      <Select
+        style={{ width: 300 }}
+        value={updatedSampleNameColumn}
+        onChange={onSampleNameColumnChange}
+        className="t-metadata-uploader-sample-name-column-select"
       >
-        {headers.map((header: string, index: number) => (
-          <BlockRadioInput key={`metadata-uploader-radio-header-${index}`}>
-            <Radio
-              key={`metadata-uploader-radio-header-${index}`}
-              value={header}
-            >
-              {header}
-            </Radio>
-          </BlockRadioInput>
+        {headers.map((header) => (
+          <Select.Option key={header.name} value={header.name}>
+            {header.name}
+          </Select.Option>
         ))}
-      </Radio.Group>
+      </Select>
+      {updatedSampleNameColumn && (
+        <Table
+          className="t-metadata-uploader-headers-table"
+          rowKey={(row) => row.rowKey}
+          columns={columns}
+          dataSource={updatedHeaders.filter(
```

Comment:
I think this should probably be in a useMemo as it should not be recalculated and here it will get recalcuated whenever a metadata restiction is updated (I think at least)

Reply:
Like this? Please see 3ac75f0abfa734bd52cf4594776fc3ba0f20c084.